src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/readytolist/ReadyToListAboutToSubmitHandler.java

```diff
@@ -2,47 +2,65 @@
 
 import static java.util.Objects.requireNonNull;
 
+import java.util.Map;
 import lombok.extern.slf4j.Slf4j;
-import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
 import uk.gov.hmcts.reform.sscs.ccd.callback.Callback;
 import uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType;
 import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
 import uk.gov.hmcts.reform.sscs.ccd.domain.*;
 import uk.gov.hmcts.reform.sscs.ccd.presubmit.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.sscs.service.RegionalProcessingCenterService;
+
 
 @Service
 @Slf4j
 public class ReadyToListAboutToSubmitHandler implements PreSubmitCallbackHandler<SscsCaseData> {
 
+    private boolean schedulingAndListingFeature;
+
+    private RegionalProcessingCenterService regionalProcessingCenterService;
+
+    public ReadyToListAboutToSubmitHandler(@Value("${feature.gaps-switchover.enabled}") boolean schedulingAndListingFeature,
```

Comment:
If you add the `@value` annotation to the field instead you don't actually need this constructor - spring will autowire fields marked as final by default.

Reply:
Ah yeah good call done this instead!